### PR TITLE
✨[source-freshcaller] add `life_cycle` to `call_metrics` stream

### PR DESCRIPTION
**Pull request by DDynamic**
## What
Include the `life_cycle` object in the call metrics stream. Remove `has_ancestry` query parameter as it is not present on the Freshcaller API. I am guessing it was leftover from the connector freshcaller was originally copied from.

https://developers.freshcaller.com/api/#call_metric_life_cycle_object

## Pre-merge Actions
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Sorry the lack of updates @DDynamic I'm going to take a look during the week to get your contribution merged.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Connector tests are failing because of incremental sync is not working properly. It produces duplicated records. Investigating..

**Comment by DDynamic** [comment]
@marcosmarxm were you able to find anything here or did you need me to take a look?

**Comment by DDynamic** [comment]
@marcosmarxm bumping

**Comment by marcosmarxm** [comment]
@DDynamic in my todo list. Hope to get updates this week.

File: airbyte-integrations/connectors/source-freshcaller/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ WORKDIR /airbyte/integration_code
 # upgrade pip to the latest version
 RUN apk --no-cache upgrade \
     && pip install --upgrade pip \
-    && apk --no-cache add tzdata build-base
+    && apk --no-cache add tzdata build-base libffi-dev
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-freshcaller/Dockerfile:10]: Why do we need `libffi-dev` here?

**Review by natikgadzhi — CHANGES_REQUESTED** [review]
Hey @DDynamic, thanks for the contribution! 

The path forward for this is: 
1. We need to merge #37160 first, that will move this connector to the standard base image and migrate from setup.py to Poetry. 
2. Then, you'd need to rebase and clean things up. It might be easier to start a new PR to be hones — schemas will be inlined in the `manifest.yaml`, so you'd need to move that change too.

File: airbyte-integrations/connectors/source-freshcaller/Dockerfile (hunk 1 above)
Comment (DDynamic) [reply airbyte-integrations/connectors/source-freshcaller/Dockerfile:10]: We shouldn't need this after migrating this connector to the standard bae image as you mentioned. I'll keep my eyes peeled for when that gets merged.

**Comment by DDynamic** [comment]
@natikgadzhi are we still waiting on https://github.com/airbytehq/airbyte/pull/38457 to merge this?

**Comment by natikgadzhi** [comment]
Thank you for the ping!

Yep, I'd prefer to ship the migration first, and we've shipped most of
them, so it should be done soon.

**Comment by DDynamic** [comment]
Hey @natikgadzhi - I just pulled from master - could I get check a ✅ now?

**Comment by natikgadzhi** [comment]
Oh hey! Kicked the tires on CI

File: airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml [hunk 2]
```diff
@@ -118,7 +116,7 @@ definitions:
       requester:
         $ref: "#/definitions/requester"
         request_parameters:
-          has_ancestry: "true"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml:121]: What was this doing? ;)

File: airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml [hunk 3]
```diff
@@ -99,8 +99,6 @@ definitions:
       $ref: "#/definitions/retriever"
       requester:
         $ref: "#/definitions/requester"
-        request_parameters:
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml:102]: Was this deprecated?

**Review by natikgadzhi — APPROVED** [review]
The changes themselves look alright, but you'd need to tweak a few things:
- In changelog you say you've added call metrics — I don't see them. Were they added previously?
- You'll need to bump `pyproject.toml` version as well, to 0.4.0 to match dockerImageTag in metadata.

File: airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml (hunk 2 above)
Comment (DDynamic) [reply airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml:121]: @natikgadzhi I don't think it was doing anything - I could find no reference on the freshcaller documentation

File: airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml (hunk 3 above)
Comment (DDynamic) [reply airbyte-integrations/connectors/source-freshcaller/source_freshcaller/manifest.yaml:102]: https://github.com/airbytehq/airbyte/pull/35892/files#r1669085305

**Comment by DDynamic** [comment]
@natikgadzhi just fixed `pyproject.toml` - sorry about that! I should have also got to your inline comments. Lastly, my changelog note is:

`✨ Source: add `life_cycle` to `call_metrics` stream`

`life_cycle` is a new field available on the call metrics stream from the Freshcaller API - just needed to pass an additional request parameter to the API to include it.

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
Hello your feedback matters a lot to us. Can you spare just 3 minutes to complete a [survey](https://form.typeform.com/to/DOBqejte)? We’re dedicated to making the contributor experience even better, and your input is key to achieving excellence. Thank you for helping us improve!

**Pull request merged**